### PR TITLE
feat(sampling): Add Sampling Limited Availability Ending alert -  [TET-390]

### PR DESCRIPTION
**Pull request by priscilawebdev**
**What this PR does:**

- Create a hook for the "Dynamic Sampling" Limited Availability program alert

We would like to inform the users participating in the availability program that their access will expire by the end of november

### CODE REVIEW

File: static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx [hunk 1]
```diff
@@ -451,6 +456,8 @@ export function ServerSideSampling({project}: Props) {
           )}
         />
 
+        <LimitedAvailabilityProgramEndingAlert organization={organization} />
```
Comment (matejminar) [inline static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:459]: Out of curiosity, how are `component:` hooks different than `feature-disabled:` hooks?
Should we use `<Feature features={['organizations:dynamic-sampling-basic', 'organizations:dynamic-sampling-advanced']} hookName="feature-disabled:dynamic-sampling-la" />` instead?

Comment (priscilawebdev) [reply static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:459]: As far as I un understood the component hook is when we wrap another component?  tbh this is not very clear to me and we do it in many different ways in the code... maybe @dashed  and @markstory can help us here?

Comment (priscilawebdev) [reply static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:459]: I copied the logic from here https://github.com/getsentry/sentry/blob/5dfe23b830db6d9150e5d2d7f3b1612fde061d71/static/app/views/organizationStats/index.tsx#L38 😄

Comment (dashed) [reply static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:459]: `HookOrDefault()` is simpler and easier to reason about here.

Comment (priscilawebdev) [reply static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:459]: we could maybe add a description to the component...something like: "hey are you wondering if you should use x or y? x is useful when...."

File: static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx [hunk 2]
```diff
@@ -498,17 +503,12 @@ export function ServerSideSampling({project}: Props) {
         />
         <TextBlock>
           {tct(
-            'Improve the accuracy of your [performanceMetrics: performance metrics] and [targetTransactions: target those transactions] which are most valuable for your organization. Server-side rules are applied immediately, with no need to re-deploy your app. To learn more about our beta program, [faqLink: visit our FAQ].',
+            'Improve the accuracy of your [performanceMetrics: performance metrics] and [targetTransactions: target those transactions] which are most valuable for your organization. Server-side rules are applied immediately, with no need to re-deploy your app.',
```
Comment (dashed) [inline static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:506]:
```suggestion
            'Improve the accuracy of your [performanceMetrics: performance metrics] and [targetTransactions: target those transactions] which are most valuable for your organization. Server-side rules are applied immediately, without having to re-deploy your app.',
```

**Review by iambriccardo — APPROVED** [review]

File: static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx [hunk 3]
```diff
@@ -70,6 +71,10 @@ import {SamplingSDKClientRateChangeAlert} from './samplingSDKClientRateChangeAle
 import {SamplingSDKUpgradesAlert} from './samplingSDKUpgradesAlert';
 import {isUniformRule, SERVER_SIDE_SAMPLING_DOC_LINK} from './utils';
 
+const LimitedAvailabilityProgramEndingAlert = HookOrDefault({
+  hookName: 'component:dynamic-sampling-limited-availability-program-ending',
+});
```
Comment (evanpurkhiser) [inline static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:76]: Why use HookOrDefault if there’s no default? Why not just <Hook />

Comment (priscilawebdev) [reply static/app/views/settings/project/server-side-sampling/serverSideSampling.tsx:76]: https://github.com/getsentry/sentry/pull/39458#discussion_r985017228

**Pull request merged**